File: delivery-service/api/src/main/java/org/delivery/api/domain/userOrder/converter/UserOrderConverter.java
package org.delivery.api.domain.userOrder.converter;

/*@Converter
public class UserOrderConverter {

    public UserOrderEntity toEntity(
            User user,
            StoreEntity storeEntity,
            List<storeMenuEntity> storeMenuEntityList
    ){
        var totalAmount = storeMenuEntityList.stream()
                .map((it -> it.getAmount()))
                .reduce(BigDecimal.ZERO*//*기본값은 0*//*, BigDecimal::add *//* 총합 *//*);

        return UserOrderEntity.builder()
                .userId(user.getId())
                .store(storeEntity)
                .amount(totalAmount)
                .build();
    }

    public UserOrderResponse toResponse(
            UserOrderEntity userOrderEntity
    ){
        return UserOrderResponse.builder()
                .id(userOrderEntity.getId())
                .orderedAt(userOrderEntity.getOrderedAt())
                .status(userOrderEntity.getStatus())
                .acceptedAt(userOrderEntity.getAcceptedAt())
                .amount(userOrderEntity.getAmount())
                .cookingStartedAt(userOrderEntity.getCookingStartedAt())
                .deliveryStartedAt(userOrderEntity.getDeliveryStartedAt())
                .receivedAt(userOrderEntity.getReceivedAt())
                .build();
    }
}*/

File: delivery-service/api/src/main/kotlin/org/delivery/api/domain/userOrder/converter/UserOrderConverter.kt
package org.delivery.api.domain.userOrder.converter

import org.delivery.api.domain.user.model.User
import org.delivery.api.domain.userOrder.controller.model.UserOrderResponse
import org.delivery.common.annotation.Converter
import org.delivery.db.store.StoreEntity
import org.delivery.db.storemenu.storeMenuEntity
import org.delivery.db.userOrder.UserOrderEntity

@Converter
class UserOrderConverter {
    fun toEntity(
        user: User?,
        storeEntity: StoreEntity?,
        storeMenuEntityList: List<storeMenuEntity>?
    ): UserOrderEntity {
        val totalMount = storeMenuEntityList
            ?.mapNotNull {
                it -> it.amount
            }?.reduce{
                 acc, bigDecimal -> acc.add(bigDecimal)
            }

        return UserOrderEntity(
            userId = user?.id,
            store = storeEntity,
            amount = totalMount
        )
    }

    fun toResponse(
        userOrderEntity: UserOrderEntity?
    ) : UserOrderResponse {
        return UserOrderResponse(
            id = userOrderEntity?.id,
            orderedAt = userOrderEntity?.orderedAt,
            status = userOrderEntity?.status,
            acceptedAt = userOrderEntity?.acceptedAt,
            amount = userOrderEntity?.amount,
            cookingStartedAt = userOrderEntity?.cookingStartedAt,
            deliveryStartedAt = userOrderEntity?.deliveryStartedAt,
            receivedAt = userOrderEntity?.receivedAt
        )
    }
}